Privacy policy

NoahMcGraw opened the pull request:
Privacy Policy generates a custom privacy policy for it's user using common clauses and allows for custom clauses to be added.
Phase 1: Established a general format for the app. Also developed the starting page and question 1 for the questionnaire.

NoahMcGraw committed:
Added transitions but requires debug from previous commit

NoahMcGraw committed:
Visual changes to banner; renaming of banner classes; Ready for pull request

NoahMcGraw committed:
Merge branch 'debug-privacyPolicy' into privacyPolicy

NoahMcGraw committed:
Fixing previous merge

NoahMcGraw committed:
[BROKEN] Merge branch 'debug-privacyPolicy' into privacyPolicy

NoahMcGraw committed:
Trying to fix broken merge from earlier
Whoopsies

NoahMcGraw committed:
Merge branch 'debug-privacyPolicy' into privacyPolicy

reesemartin reviewed (changes requested):
You have a lot of stuff in the controller that should be in partials. You should keep the controller just manipulating the data and put all that text in a partial.
You should create directives to split all that up that way you arent mixing visual stuff with the controller and so that your index.php doesnt become this super long file.

Should do something like:
```
<policy-initial-info ng-show="newPolicy.addingInfo"></policy-initial-info>
<policy-questionaire ng-show="newPolicy.questionaire"></policy-questionaire>
```
And then inside those call sub directives for each page in the questionaire.


Doing directives will also make it easier to just drop this into a wordpress plugin format instead of an html file like it is. So create a directive for the app as a whole even above the initial info and questionaire.


Also all your inputs with validation like required need ng-messages.
```
<div ng-messages="initialInfoForm.companyName.$error">
    <div ng-message="required"><span class="error">This field is required.</span></div>
</div>
```


Remove the unisyn footer module

NoahMcGraw committed:
Added .gitignore file

NoahMcGraw committed:
Removed unisynFooter; Added validation for required inputs in initialInfo form;
Divided app into directives; logic previously held in ctrl for page changes
is now changed to tabs controlling individual directives that each hold a
question.

NoahMcGraw committed:
Increased main div size to 1000px; Changed answer key for questionnaires to
string false rather than bool;

NoahMcGraw committed:
Removed questionnaire ctrl(not neccesary) and moved contents to main ctrl;
Changed size of main div to 900px; Removed ng-disabled from Generate Policy
button for time being;

reesemartin commented on app/controllers/main-ctrl.js:
```diff
@@ -14,104 +14,36 @@ app.controller('mainCtrl', ['$scope', function($scope){
-                },
-            ],
-            question: "Do you track any other forms of user data?",
+    $scope.questionnaire = {
```
What does this object do? I was thinking it was unlikely you should be doing "false" instead of the boolean false so I looked for where its being used and cant find anything.

NoahMcGraw replied:
This object holds the results from each questionnaire tab. The answer key is being used as a model in the md-radio-group at the bottom of each tab. As far as I have been able to tell, the value passed in md-radio is always passed as a string. I was having issues with getting the value to bind to the ctrl as a bool true or false but I think it has everything to do with me using "value" instead of "ng-value". Will test and follow up on that.

NoahMcGraw replied:
Yep that's what it was. I've made the changes to reflect that logic in the ctrl and partial

reesemartin commented on app/directives/policyQuestionnaire.js:
```diff
@@ -0,0 +1,8 @@
+    var directive = {
+        restrict: 'E',
+        templateUrl: '/app/partials/policyQuestionnaire-template.php',
+        controller: 'mainCtrl',
```
Not necessary since this directive does not create a child scope, it just uses whatever the parent is using which is already mainCtrl

reesemartin commented on app/directives/questionnaireTabs/questionnaireSharing.js:
```diff
@@ -0,0 +1,8 @@
+    var directive = {
+        restrict: 'E',
+        templateUrl: '/app/partials/questionnaireTabs/questionnaireSharing-template.php',
+        controller: 'mainCtrl',
```
Not necessary since this directive does not create a child scope, it just uses whatever the parent is using which is already mainCtrl

reesemartin commented on app/directives/questionnaireTabs/questionnaireStorage.js:
```diff
@@ -0,0 +1,8 @@
+    var directive = {
+        restrict: 'E',
+        templateUrl: '/app/partials/questionnaireTabs/questionnaireStorage-template.php',
+        controller: 'mainCtrl',
```
Not necessary since this directive does not create a child scope, it just uses whatever the parent is using which is already mainCtrl

reesemartin commented on app/directives/questionnaireTabs/questionnaireUsage.js:
```diff
@@ -0,0 +1,8 @@
+    var directive = {
+        restrict: 'E',
+        templateUrl: '/app/partials/questionnaireTabs/questionnaireUsage-template.php',
+        controller: 'mainCtrl',
```
Not necessary since this directive does not create a child scope, it just uses whatever the parent is using which is already mainCtrl

reesemartin commented on app/partials/policyQuestionnaire-template.php:
```diff
@@ -0,0 +1,17 @@
+            <questionnaire-sharing></questionnaire-sharing>
+        </md-tab>
+    </md-tabs>
+    <md-button class="md-primary md-raised">
```
I think it would help the UX if you added a css rule like this:
```
.unisyn-big-button.md-button {
    width: 100%;
    padding-top: 10px;
    padding-bottom: 10px;
    margin-left: 0;
    margin-right: 0;
    font-size: 16px;
}
```
Then use class unisyn-big-button on this button to make it more obvious down there.

reesemartin commented on app/partials/policyQuestionnaire-template.php:
```diff
@@ -0,0 +1,17 @@
+    <md-button class="md-primary md-raised">
+        Generate Policy
+    </md-button>
+    <md-button class="md-accent" ng-click="backToInitialInfo()">Edit Company Info</md-button>
```
Might be nice to move this to the top right of the toolbar in the questionaire tabs and remove the md-accent so its just white text in the toolbar like this:
![image](https://user-images.githubusercontent.com/19984334/42108835-447eddb2-7ba1-11e8-82e7-c38788e87b8d.png)

reesemartin commented on app/partials/questionnaireTabs/questionnaireSharing-template.php:
```diff
@@ -0,0 +1,31 @@
+<md-toolbar>
+        <div class="md-toolbar-tools" style="text-align: center;">
```
Remove the text-align: center here once you have added the edit company info button to the toolbar.

reesemartin commented on app/partials/privacyPolicyGenerator-template.php:
```diff
@@ -0,0 +1,25 @@
+                you can have a custom policy on your desk in as little as <strong>Five Minutes</strong>!
+                </p>
+            </div>
+            <md-button class="md-primary md-raised" ng-click="startNewPolicy()" ng-hide="newPolicy.addingInfo || newPolicy.questionaire">Let's Get Started</md-button>
```
Same as the generate policy button, I think it would help the UX if you added a css rule like this:
```
.unisyn-big-button.md-button {
    width: 100%;
    padding-top: 10px;
    padding-bottom: 10px;
    margin-left: 0;
    margin-right: 0;
    font-size: 16px;
}
```
Then use class unisyn-big-button on this button

reesemartin commented on index.php:
```diff
@@ -1,144 +1,34 @@
+        <script src="/app/app.js"></script>
+        <!-- Controllers -->
+        <script src="/app/controllers/main-ctrl.js"></script>
+        <script src="/app/controllers/policyQuestionnaire-ctrl.js"></script>
```
This file is a 404 so remove the link

reesemartin commented on app/partials/policyQuestionnaire-template.php:
```diff
@@ -0,0 +1,17 @@
+<md-content class="md-block md-padding md-vertical-margin privacyPolicyGeneratorContainer"                          ng-show="newPolicy.questionaire">
+    <md-tabs md-dynamic-height="true">
```
I suggest using
```
md-selected="newPolicy.selectedTab"
```
here to also be able to programatically change tabs.

In your controller add that property to newPolicy at the top and initialize it as 0. It works like an array with 0 being first tab 1 being second.

reesemartin commented on app/partials/policyQuestionnaire-template.php:
```diff
@@ -0,0 +1,17 @@
+        <md-tab label="Data Sharing">
+            <questionnaire-sharing></questionnaire-sharing>
+        </md-tab>
+    </md-tabs>
```
I would add a button above generate policy for "Next" that increments newPolicy.selectedTab by 1. If newPolicy.selectedTab equals 2 then it is on the last tab so hide the button.

No real need for a back button since they can click tabs at top to go back.

I would apply md-accent and md-raised and the unisyn-big-button classes.

This isnt required but I think it would help. If they messed up or want to review something they can break the flow by scrolling up and selecting a tab to review or clicking edit company info. Otherwise they would fill out the info moving down and then click next or generate.

reesemartin reviewed (changes requested):
All the templates should be html files not php files. There could be some neat server interaction using php files as templates. I didnt know you could do that and it work. But this will not need it so change them to html and the template urls in directives please.

NoahMcGraw committed:
Changes requested for Phase 1 of Privacy Policy:

Changed md-radio buttons values to bool true false using ng-value and
reflected changes in ctrl and other partials;

NoahMcGraw committed:
Changes requested for Phase 1 of Privacy Policy Generator:

Changed file type for templates to html and changed their respective paths
in the directive files;

md-radio-buttons now use ng-value and hold bool
true false values;

Removed controller key when defining directives in their
respective js files;

 Created unisyn-big-button class and implemented it
with most buttons in app;

Moved "Edit Company Info" button to header of each tab and aligned it properly;

Removed unused controller link in index head;

Added model to newPolicy to watch the value of the selected tab and manipulate
it in the ctrl;

Added "Next" button to questionnaire;

NoahMcGraw committed:
Added remaining tabs for questionnaire; Added description dialog for
legitimate interest in data usage.

NoahMcGraw committed:
added close icon to dialog box in data usage tab and added

NoahMcGraw committed:
Added close icon for dialog in data usage tab and added cancelDialog function
to ctrl;

reesemartin commented on index.php:
```diff
@@ -1,144 +1,33 @@
+        <meta charset="UTF-8">
+        <title>Unisyn Privacy Policy Generator</title>
+        <link rel="shortcut icon" type="image/x-icon" href="/app/assets/images/icons/unisynFavicon.png">
+        <link href="/app/modules/angular-material/1.1.5/angular-material.min.css" type="text/css" rel="stylesheet">
```
Add material icons stylesheet here to gain access to them in the app
```
<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```

reesemartin commented on app/partials/questionnaireTabs/questionnaireUsage-template.html:
```diff
@@ -0,0 +1,97 @@
+                <div class="md-toolbar-tools">
+                    <label class="md-headline" flex>Examples of Legitimate Interest</label>
+                    <md-button class="md-icon-button" ng-click="cancelDialog()">
+                        <md-icon md-svg-src="/app/assets/images/icons/close-24px.svg">
```
Use material-icons font set to call the material icon
```
<md-icon md-font-set="material-icons"> close </md-icon>
```

reesemartin commented on app/assets/images/icons/close-24px.svg:
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
```
Delete this file

reesemartin commented on app/assets/images/icons/help-24px.svg:
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
```
Delete this file

reesemartin commented on app/partials/questionnaireTabs/questionnaireUsage-template.html:
```diff
@@ -0,0 +1,97 @@
+    <strong><em>"Legitimate Interest"</em></strong>
+    <md-button class="md-icon-button" ng-click="openLegitimateInterestDialog($event)">
+        <md-tooltip md-direction="right">What Is This?</md-tooltip>
+        <md-icon md-svg-src="/app/assets/images/icons/help-24px.svg">
```
Use material-icons font set to call the material icon
```
<md-icon md-font-set="material-icons"> help </md-icon>
```

NoahMcGraw committed:
Added link to md-icon-provider in header and changed icons in the app to use
the material icons set; Deleted local files for previously mentioned icons;
Deleted duplicate banner img in assets folder,
